commands/moderation/tempban.js

```diff
@@ -0,0 +1,105 @@
+const Discord = require('discord.js');
+const ms = require('ms');
+var dateFormat = require('dateformat');
+
+module.exports = {
+    name: "tempban",
+    description: "Temporarily ban a user",
+    
+    execute(msg, args, con) {
+        const user_information = /(\<@!?\d+\>)\s(\d+[yhwdms])\s(.+)$/
+
+        if (!msg.member.roles.cache.some(
+            role => role.name === "Admin" || role.name === "Moderator")) {
+                return msg.reply("You must be an Admin to use this command.");
+        }else{
+
+            if(!args.join(" ").match(user_information)) {
+                return msg.reply("The command you sent isn't in a valid format")
+            }
+
+            const [, id, timeLength, reason] = args.join(" ").match(user_information) ?? []
+
+            const toTempBan = msg.mentions.members.first();
+            if (!toTempBan) {
+                return msg.reply("Please provide a user to temporarily ban.");
+            }
+
+            if(toTempBan.id == msg.author.id) {
+                return msg.reply("You can't temporarily ban yourself!")
+            }
+
+            if(toTempBan.hasPermission('BAN_MEMBERS')) {
+                return msg.reply("This user also has ban privileges.")
+            }            
+            
+            // const reason = args.slice(args.indexOf(timeLength[0]) + 1).join(" ")
+            console.log(reason)
+            if (reason === "") {
+                return msg.reply("Please provide a reason for temporarily banning this user.");
+            }
+
+            let now = new Date();
+            let date = dateFormat(now, "yyyy-mm-dd HH:MM:ss");
+            const action = "cc!tempban " + args.join(" ")
+            // Inserts row into database
+            var sql = `INSERT INTO infractions (timestamp, user, action, length_of_time, reason, valid, moderator) VALUES
+            ('${date}', '${toTempBan.id}', 'cc!tempban', '${timeLength}', '${reason}', true, '${msg.author.id}');
+            INSERT INTO mod_log (timestamp, moderator, action, length_of_time, reason) VALUES
+            ('${date}', '${msg.author.id}', '${action}', '${timeLength}', '${reason}');`;
```

Comment:
@selectlearns brought up a great point in Discord that we might want to escape these values while we wait for the ORM to get implemented.

You can find documentation on how to do it [here](https://www.npmjs.com/package/mysql#escaping-query-values), and Select showed an example in Discord as well.

Using this method in our SQL statements should fend off SQLi attacks for now and also prevent problems like if we use apostrophes in our reasons, etc. (apostrophes break the INSERT statements)

Reply:
Makes sense, makes sense, I'll do this while I refactor the functions